src/update_agent/actor.rs

```diff
@@ -367,29 +357,22 @@ impl UpdateAgentInfo {
                 log::trace!("{}", msg);
             }
         };
-
-        Ok(())
     }
 
     /// Try to finalize an update.
-    async fn tick_finalize_update(
-        &self,
-        state: &mut UpdateAgentState,
-        release: Release,
-    ) -> Result<(), ()> {
+    async fn tick_finalize_update(&self, state: &mut UpdateAgentState, release: Release) {
         trace!("trying to finalize an update");
         FINALIZATION_ATTEMPTS.inc();
 
         let strategy_can_finalize = self.strategy.can_finalize().await;
-        let state_change = if !strategy_can_finalize {
+        if !strategy_can_finalize {
             utils::update_unit_status(&format!(
                 "update staged: {}; reboot pending due to update strategy",
                 &release.version
             ));
             // Reset number of postponements to `MAX_FINALIZE_POSTPONEMENTS`
             // if strategy does not allow finalization.
             state.update_staged(release);
-            Err(())
```

Comment:
As we are not throwing back errors from all branches anymore, it could be worth adding a direct early return here and in the `else` branch below, so that the bottom part of this function/logic is not anymore deeply nested into multiple if-else. I suspect the resulting code layout would be easier to read.